x/superfluid/types/expected_keepers.go

```diff
@@ -47,6 +48,9 @@ type GammKeeper interface {
 	GetPoolAndPoke(ctx sdk.Context, poolId uint64) (gammtypes.CFMMPoolI, error)
 	GetPoolsAndPoke(ctx sdk.Context) (res []gammtypes.CFMMPoolI, err error)
 	ExitPool(ctx sdk.Context, sender sdk.AccAddress, poolId uint64, shareInAmount sdk.Int, tokenOutMins sdk.Coins) (exitCoins sdk.Coins, err error)
+	GetMigrationInfo(ctx sdk.Context) gammtypes.MigrationRecords
+	Migrate(ctx sdk.Context, sender sdk.AccAddress, sharesToMigrate sdk.Coin) (amount0, amount1 sdk.Int, liquidity sdk.Dec, poolIdLeaving, poolIdEntering uint64, err error)
```

Comment:
Why do we need gamm keeper's Migrate method?

Reply:
We don't, good catch! Before I wanted to make superfluid the only entrypoint and simply skip the unlocking part if it wasn't locked, but it ended up being a bad design and I scrapped it but left this behind. Removed.